infra/gp-hashicorp-vault/templates/Monitoring.yaml

```diff
@@ -0,0 +1,53 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: prometheus-k8s
+  namespace: {{ .Release.Namespace }}
+rules:
+  - verbs:
+      - get
+      - list
+      - watch
+    apiGroups:
+      - ''
+    resources:
+      - services
+      - endpoints
+      - pods
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: prometheus-k8s
+  namespace: {{ .Release.Namespace }}
+subjects:
+  - kind: ServiceAccount
+    name: prometheus-k8s
+    namespace: openshift-monitoring
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: prometheus-k8s
+---
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: vault-metrics
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "gp-hashicorp-vault.labels" . | nindent 4 }}
+spec:
+  podMetricsEndpoints:
+    - interval: 2m
+      params:
+        format:
+          - prometheus
+      path: /v1/sys/metrics
+      port: http
+      scheme: http
+  namespaceSelector:
```

Comment:
Die Angabe können wir uns sparen da automatisch nur in diesem Namespace gescraped wird